docs/serving/samples/hello-world/helloworld-shell/index.md

```diff
@@ -5,4 +5,195 @@ weight: 1
+  and a Docker Hub account configured.
+- (optional) The Knatice CLI client [kn](https://github.com/knative/client/releases) that simplifies the deployment. Alternative you can also use `kubectl` and apply resource files directly.
+
+## Build
+
+1. Create a new file named `script.sh` and paste the script below. This will run netcat (`nc`) in an endless loop, returning how friendly welcome message.
+
+  ```shell
+  #!/bin/sh
+  while true ; do
+    echo -e "HTTP/1.1 200\n\n ${TARGET:=World}!\n" | nc -l -p 8080 -q 1;
+  done
+  ```
+
+1. Create a new file named `Dockerfile` and copy the code block below into it.
+
+   ```docker
+   # Use the official Alpine image for a lean production container.
+   # https://hub.docker.com/_/alpine
+   FROM alpine:3
+
+   # Install netcat (nc)
+   RUN apk update \
+    && apk add netcat-openbsd
+
+   # Copy over the service script
+   COPY script.sh /
+
+   # Start up the webserver
+   CMD [ "/bin/sh", "/script.sh" ]
+   ```
+
+Once you have recreated the sample code files (or used the files in the sample
+folder) you're ready to build and deploy the sample app.
+
+1. Use Docker to build the sample code into a container. To build and push with
+   Docker Hub, run these commands replacing `{username}` with your Docker Hub
+   username:
+
+   ```shell
+   # Build the container on your local machine
+   docker build -t {username}/helloworld-shell .
+
+   # Push the container to docker registry
+   docker push {username}/helloworld-shell
+   ```
+
+## Deploying
+
+1. After the build has completed and the container is pushed to docker hub, you
+   can deploy the app into your cluster.
+
+   {{< tabs name="helloworld_shell" default="kn" >}}
+   {{% tab name="yaml" %}}
+
+   1. Create a new file, `service.yaml` and copy the following service definition
+      into the file. Make sure to replace `{username}` with your Docker Hub
+      username.
+
+      ```yaml
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      metadata:
+        name: helloworld-shell
+        namespace: default
+      spec:
+        template:
+          spec:
+            containers:
+              - image: docker.io/{username}/helloworld-shell
+                env:
+                  - name: TARGET
+                    value: "Shell Sample v1"
+      ```
+
+   Ensure that the container image value
+   in `service.yaml` matches the container you built in the previous step. Apply
+   the configuration using `kubectl`:
+
+   ```shell
+   kubectl apply --filename service.yaml
+   ```
+
+   {{< /tab >}}
+   {{% tab name="kn" %}}
+
+   With `kn` you can deploy the service with
+
+   ```shell
+   kn service create helloworld-shell --image=docker.io/{username}/helloworld-shell --env TARGET="Go Sample v1"
+   ```
+
+   This will wait until your service is deployed and ready, and ultimately it will print the URL through which you can access the service.
+
+   The output will look like in
```

Comment:
should we add output of the kn service create

Reply:
ah, totally forget to c&p the output of the command. thanks for the heads up!